.github/workflows/ci.yml

```diff
@@ -0,0 +1,42 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java#apache-maven-with-a-settings-path
+
+name: Maven Package and Publish
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Replace Macros
+        run: |
+          grep -rl "__PUBLISHER__" settings.xml | xargs sed -i s@"__PUBLISHER__"@"$PUBLISHER"@g
+          grep -rl "__TOKEN__" settings.xml | xargs sed -i s@"__TOKEN__"@"$TOKEN"@g
+        env:
+          PUBLISHER: ${{ github.actor }}
+          TOKEN: ${{ github.token }}
+
+      - name: Compile with Maven
```

Comment:
We'll need to integrate this into winch to provide version bumping etc

Reply:
Yes, that is next on the list to do

Reply:
Created a PR with that and also updated winch with your commands.